src/main/java/org/reaktivity/nukleus/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -1328,6 +1367,16 @@ private void onDecodeConnect(
                     this.topicAliasMaximum = (short) Math.min(topicAliasMaximum, topicAliasMaximumLimit);
                     break;
                 case KIND_SESSION_EXPIRY:
+                    if (sessionExpiryInterval != -1)
+                    {
+                        sessionExpiryInterval = 0;
+                        reasonCode = PROTOCOL_ERROR;
+                        break decode;
+                    }
+                    final int sessionExpiryInterval = mqttProperty.expiryInterval();
+                    this.sessionExpiryInterval = Math.min(sessionExpiryInterval, sessionExpiryIntervalLimit);
+                    this.sessionExpiresAt = System.currentTimeMillis() + TimeUnit.SECONDS.toMillis(sessionExpiryInterval);
```

Comment:
Given that the session expires a N seconds after the client disconnects abruptly, where session expiry interval is N, is there value to keeping sessionExpiresAt field state in memory?